src/Widgets/SaveDialog.vala

```diff
@@ -73,6 +73,16 @@ public class Screenshot.SaveDialog : Gtk.Dialog {
         preview_box_context.add_class (Granite.STYLE_CLASS_CARD);
         preview_box_context.add_class (Granite.STYLE_CLASS_CHECKERBOARD);
 
+        var preview_event_box = new Gtk.EventBox ();
+        preview_event_box.add (preview_box);
+
+        Gtk.drag_source_set (preview_event_box, Gdk.ModifierType.BUTTON1_MASK, null, Gdk.DragAction.COPY);
+        Gtk.drag_source_add_image_targets (preview_event_box);
+        Gtk.drag_source_set_icon_gicon (preview_event_box, preview.gicon);
```

Comment:
Probably better to use a system icon here since the screenshot image is fairly large
```suggestion
        Gtk.drag_source_set_icon_gicon (preview_event_box, new ThemedIcon("image-x-generic"));
```